tests/run/pure_pxd.srctree

```diff
@@ -55,6 +55,13 @@ def func(a, b, c):
     """
     return a + b + c
 
+def sum_generator_expression(a):
+    """
+    >>> sum_generator_expression(5)
+    10
+    """
+    return sum(i for i in range(a))
```

Comment:
I don't think this test is actually run. Since it's a `cdef` function it isn't available directly from doctest. The solution is to add a def function that calls it
```suggestion
def sum_generator_expression(a):
    # GH-3477 - closure variables incorrectly captured in functions transformed to cdef
    return sum(i for i in range(a))

def run_sum_generator_expression(a):
    """
    >>> sum_generator_expression(5)
    10
    """
    return sum_generator_expression(a)
```